03_python_fastapi_project/main.py

```
@@ -144,6 +162,61 @@ async def get_product(product_id: int, db: AsyncSession = Depends(get_db)):
     return product_to_response(db_product)
 
 
+@app.post("/cart/add/{product_id}", response_model=CartItemResponse)
+async def add_to_cart(product_id: int, db: AsyncSession = Depends(get_db)):
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)